sdk/runanywhere-kotlin/src/jvmMain/kotlin/com/runanywhere/sdk/storage/JvmSecureStorage.kt

```
@@ -48,35 +70,135 @@ internal class JvmSecureStorage : SecureStorage {
+    /**
+     * Sets secure file permissions (600 - owner read/write only)
+     */
+    private fun setSecureFilePermissions(file: File) {
+        try {
+            val path = Paths.get(file.absolutePath)
+            val permissions = setOf(
+                PosixFilePermission.OWNER_READ,
+                PosixFilePermission.OWNER_WRITE
+            )
+            Files.setPosixFilePermissions(path, permissions)
+        } catch (e: UnsupportedOperationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)